portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -0,0 +1,76 @@
+package com.google.sps.servlets;
+
+import com.google.sps.data.Marker;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+/** Handles fetching and saving markers data. */
+@WebServlet("/markers")
+public class MarkerServlet extends HttpServlet {
+
+    /** Responds with a JSON array containing marker data. */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json");
+
+        Collection<Marker> markers = getMarkers();
+        Gson gson = new Gson();
+        String json = gson.toJson(markers);
+
+        response.getWriter().println(json);
+    }
+
+    /** Accepts a POST request containing a new marker. */
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) {
+        double lat = Double.parseDouble(request.getParameter("lat"));
+        double lng = Double.parseDouble(request.getParameter("lng"));
+        String content = Jsoup.clean(request.getParameter("content"), Whitelist.none());
```

Comment:
Great. Where did you hear about Jsoup?

Reply:
The walkthrough provided us an example using Jsoup, which I followed for the map marker portion!